From: Danny Feng <dfeng@redhat.com>
Date: Fri, 9 Oct 2009 10:22:57 -0400
Subject: [net] vlan: silence multicast debug messages
Message-id: <4ACF0F01.1060701@redhat.com>
Patchwork-id: 21074
O-Subject: Re: [PATCH RHEL5.5] vlan: silence multicast debug messages
Bugzilla: 461442
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Andy Gospodarek <gospo@redhat.com>

RHBZ #:
https://bugzilla.redhat.com/show_bug.cgi?id=461442

Description:
If one subscribes to a few hundred multicasts on one or more
VLAN interfaces, the VLAN driver logs an excessive number of
messages and wipes out the more useful information in the
kernel message log that is viewed with the 'dmesg' command.

Since we can observe mcast via '/proc/net/dev_mcast', put
those output into pr_debug.

diff --git a/net/8021q/vlan_dev.c b/net/8021q/vlan_dev.c
index 15b5719..2c7a5f1 100644
--- a/net/8021q/vlan_dev.c
+++ b/net/8021q/vlan_dev.c
@@ -771,7 +771,7 @@ static void vlan_flush_mc_list(struct net_device *dev)
 	struct dev_mc_list *dmi = dev->mc_list;
 
 	while (dmi) {
-		printk(KERN_DEBUG "%s: del %.2x:%.2x:%.2x:%.2x:%.2x:%.2x mcast address from vlan interface\n",
+		pr_debug("%s: del %.2x:%.2x:%.2x:%.2x:%.2x:%.2x mcast address from vlan interface\n",
 		       dev->name,
 		       dmi->dmi_addr[0],
 		       dmi->dmi_addr[1],
@@ -862,7 +862,7 @@ void vlan_dev_set_multicast_list(struct net_device *vlan_dev)
 		for (dmi = vlan_dev->mc_list; dmi != NULL; dmi = dmi->next) {
 			if (vlan_should_add_mc(dmi, VLAN_DEV_INFO(vlan_dev)->old_mc_list)) {
 				dev_mc_add(real_dev, dmi->dmi_addr, dmi->dmi_addrlen, 0);
-				printk(KERN_DEBUG "%s: add %.2x:%.2x:%.2x:%.2x:%.2x:%.2x mcast address to master interface\n",
+				pr_debug("%s: add %.2x:%.2x:%.2x:%.2x:%.2x:%.2x mcast address to master interface\n",
 				       vlan_dev->name,
 				       dmi->dmi_addr[0],
 				       dmi->dmi_addr[1],
@@ -880,7 +880,7 @@ void vlan_dev_set_multicast_list(struct net_device *vlan_dev)
 				 * delete it from the real list on the underlying device.
 				 */
 				dev_mc_delete(real_dev, dmi->dmi_addr, dmi->dmi_addrlen, 0);
-				printk(KERN_DEBUG "%s: del %.2x:%.2x:%.2x:%.2x:%.2x:%.2x mcast address from master interface\n",
+				pr_debug("%s: del %.2x:%.2x:%.2x:%.2x:%.2x:%.2x mcast address from master interface\n",
 				       vlan_dev->name,
 				       dmi->dmi_addr[0],
 				       dmi->dmi_addr[1],
